pelita/maze_generator.py

```diff
@@ -252,53 +289,80 @@ def generate_half_maze(width, height, ngaps_center, bots_pos, rng=None):
     # use binary space partitioning
     rng = default_rng(rng)
 
-    # outer walls are top, bottom, left and right edge
-    walls = {(x, 0) for x in range(width)} | \
-            {(x, height-1) for x in range(width)} | \
-            {(0, y) for y in range(height)} | \
-            {(width-1, y) for y in range(height)}
+    # outer walls except the border
+    walls = (
+        # top
+        {(x, 0) for x in range(width // 2)}
+        # bottom
+        | {(x, height - 1) for x in range(width // 2)}
+        # left
+        | {(0, y) for y in range(height)}
+    )
 
-    # Generate a wall with gaps at the border between the two homezones
+    #
+    # BORDER SAMPLING
+    #
+    # generate a wall with gaps at the border between the two homezones
     # in the left side of the maze
 
-    # TODO: when we decide to break backward compatibility with the numpy version
-    # of create maze, this part can be delegated directly to generate_walls and
-    # then we need to rewrite mirror to mirror a set of coordinates around the center
-    # by discarding the lower part of the border
+    # start with a full wall at the left side of the border
+    pos = width // 2 - 1
+    border = {(pos, y) for y in range(1, height - 1)}
 
-    # Let us start with a full wall at the left side of the border
-    x_wall = width//2 - 1
-    wall = {(x_wall, y) for y in range(1, height - 1)}
-
-    # possible locations for gaps
+    # possible locations for gaps;
     # these gaps need to be symmetric around the center
-    # TODO: when we decide to break compatibility with the numpy version of
-    # create_maze we can rewrite this. See generate_walls for an example
+    #
+    # TODO:
+    # when we drop compatibility with numpy mazes, this might be rewritten to
+    # sample wall segments to keep with k = len(wall) - ngaps
     ymax = (height - 2) // 2
     candidates = list(range(ymax))
-    rng.shuffle(candidates)
+    candidates = sample(candidates, ngaps_center//2, rng)
+
+    # save gaps and edges for chamber finding
+    gaps = set()
+    edges = set()
+
+    # remove gaps from top and mirrored from bottom
+    for y in candidates:
+        upper = (pos, y + 1)
+        lower = (pos, height - 2 - y)
+
+        # add both gaps
+        gaps.add(upper)
+        gaps.add(lower)
+
+        # edges between those gaps which would be connected after mirroring
+        edges.add((upper, lower))
```

Comment:
I like this trick, but I am afraid this will be very confusing in the long run. Maybe it is enough to find a better name for this? After all these are "fake" edges, in the sense that they are not going to be there once we have the full maze, right? It should be clear that this here is a trick.